app/dashboard/static/js/app/view-socs-soc-job-kernel.2017.4.js

```diff
@@ -20,654 +23,483 @@
+    }
 
-            html.replaceContent(
-                document.getElementById('git-describe'), domNode);
-
-            // Git URL.
-            if (gitURLs[0]) {
-                aNode = document.createElement('a');
-                aNode.setAttribute('href', gitURLs[0]);
-                aNode.appendChild(document.createTextNode(gitURL));
-                aNode.insertAdjacentHTML('beforeend', '&nbsp;');
-                aNode.appendChild(html.external());
-            } else {
-                if (gitURL && gitURL !== undefined) {
-                    aNode = document.createTextNode(gitURL);
-                } else {
-                    aNode = html.nonavail();
-                }
-            }
-            html.replaceContent(document.getElementById('git-url'), aNode);
-
-            // Git commit.
-            if (gitURLs[1]) {
-                aNode = document.createElement('a');
-                aNode.setAttribute('href', gitURLs[1]);
-                aNode.appendChild(document.createTextNode(gitCommit));
-                aNode.insertAdjacentHTML('beforeend', '&nbsp;');
-                aNode.appendChild(html.external());
-            } else {
-                if (gitCommit && gitCommit !== null) {
-                    aNode = document.createTextNode(gitCommit);
-                } else {
-                    aNode = html.nonavail();
-                }
-            }
-            html.replaceContent(document.getElementById('git-commit'), aNode);
+    function getPlansFailed() {
+        detailsFailed();
+        plansFailed();
+    }
 
-            // Date.
-            domNode = document.createElement('time');
-            domNode.setAttribute('datetime', createdOn.toISOString());
-            domNode.appendChild(
-                document.createTextNode(createdOn.toCustomISODate()));
-            html.replaceContent(
-                document.getElementById('job-date'), domNode);
+    function getPlansDone(response) {
+        if (response.result.length === 0) {
+            getPlansFailed();
+            return
         }
+    
+        updateDetails(response.result[0]);
+        updatePlansTable(response.result);
+        getBatchCount(response.result);
+        getBatchStatus(response.result);
+
+        gPlansTable
+            .search(gSearchFilter);
     }
 
-    function getBootDoneChart(response) {
-        chart.bootpie({
-            element: 'boot-chart',
-            response: response,
-            legend: true
-        });
+    function getPlans() {
+        var data;
+        var deferred;
+
+        data = {
+            aggregate: 'name',
+            job: gJob,
+            kernel: gKernel,
+            mach: gSoc,
+            parent_id: 'null',
+            sort: 'name',
+            field: [
+                'name',
+                'created_on',
+                'job',
+                'git_branch',
+                'git_commit',
+                'git_url',
+                'kernel',
```

Comment:
Maybe add `mach` here to have it in `updatePlansTable()` and fix the URLs?

Reply:
updated